File: yujung/프로그래머스/의상.cpp
#include <string>
#include <vector>
#include<map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string,int> m;
    for(auto a: clothes){
        m[a[1]]++;
    }
    for(auto i: m){
        answer*=(++i.second);
    }
    return answer-1;
}

File: seoyeon/프로그래머스/알고리즘 고득점 Kit/해시/의상.py
from collections import defaultdict

def solution(clothes):
    answer = 1
    
    dict = defaultdict(list)
    
    for cloth,type in clothes:
        dict[type].append(cloth)

    n_lst = [] #dict 개수
    for d in dict:
        n_lst.append(len(dict[d]))
        
    #선택하거나 안 하거나
    for i in range(len(n_lst)):
        answer *= (n_lst[i]+1) #가능한 경우의 수(n_lst[i]개) + 아무것도 선택하지 않는 경우 
    
    answer -= 1 #아무것도 선택되지 않는 경우
    
    return answer